src/main/java/com/cloudbees/hudson/plugins/folder/ChildNameGenerator.java

```diff
@@ -325,6 +292,7 @@ final ResultWithOptionalSave<File> ensureItemDirectory(@NonNull P parent, @NonNu
      * @param item the item.
      * @param legacyDirName the name of the directory that the item was loaded from.
      * @throws IOException if the ideal name could not be attached to the item.
+     * @deprecated removed without replacement
```

Comment:
This method is no longer used as of this PR. I could delete it if desired, but that would cause a compile-time error in `branch-api` once updating `branch-api` to a BOM version that contains this PR.

Reply:
IIUC the overrides in `branch-api` can be deleted at any time, at which point we could also delete this method.

Reply:
Yes, exactly.

Reply:
⇒ https://github.com/jenkinsci/branch-api-plugin/pull/510